web-services/query/src/main/java/datawave/webservice/query/logic/filtered/QueryLogicFilterByAuth.java

```diff
@@ -0,0 +1,101 @@
+package datawave.webservice.query.logic.filtered;
+
+import datawave.webservice.query.Query;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.core.security.ColumnVisibility;
+import org.apache.accumulo.core.security.VisibilityEvaluator;
+import org.apache.accumulo.core.security.VisibilityParseException;
+
+import java.util.Set;
+
+public class QueryLogicFilterByAuth implements FilteredQueryLogic.QueryLogicFilter {
+    
+    public static enum MatchType {
+        // match only the first of the Authorizations (presumably the user)
+        FIRST,
+        // match all of the Authorizations (the user and proxies)
+        ALL,
+    }
+    
+    // A visibility string to be matched against the auths being used for the query
+    private ColumnVisibility visibility;
+    
+    // Default is to only require the user visibilities to match
+    private MatchType matchType = MatchType.ALL;
+    
+    // if negated than the negation of the match is returned
+    private boolean negated = false;
+    
+    public QueryLogicFilterByAuth() {}
+    
+    public QueryLogicFilterByAuth(String visibility) {
+        setVisibility(visibility);
+    }
+    
+    public QueryLogicFilterByAuth(String visibility, MatchType matchType) {
+        this(visibility);
+        setMatchType(matchType);
+    }
+    
+    public QueryLogicFilterByAuth(String visibility, MatchType matchType, boolean negated) {
+        this(visibility, matchType);
+        setNegated(negated);
+    }
+    
+    @Override
+    public boolean canRunQuery(Query settings, Set<Authorizations> auths) {
+        boolean canRunQuery = matches(auths);
+        if (negated) {
+            canRunQuery = !canRunQuery;
+        }
+        return canRunQuery;
+    }
+    
+    private boolean matches(Set<Authorizations> auths) {
```

Comment:
ooh, I could really use the ability to do this 'generically', outside of the context of a `QueryLogicFilter`, can we implement the auth/visibility matching logic in a utility class that could be reused?

My use case is - looking at events that are being returned to the end-user and determining whether they need to be encrypted based on the presence of certain elements in the ColumnVisibility.

Reply:
Let me see what I can do....

Reply:
Creating java.util.Predicate implementations that you can reuse.